presto-spi/src/main/java/com/facebook/presto/spi/type/TypeSignature.java

```diff
@@ -324,6 +323,13 @@ else if (literalCalculationParameters.contains(parameterName)) {
         }
     }
 
+    private static boolean isValidStartOfIdentifier(char c)
+    {
+        return (c >= 'a' && c <= 'z') ||
+                (c >= 'A' && c <= 'Z') ||
```

Comment:
Previously, `isAlphabetic(c)` was used. For ASCII, this is equivalent. Are we limited to ASCII?

Why not just `isAlphabetic(c) || c == '_'`?

Reply:
Since `parseTypeSignature` is a public static method, there is no guarantee that caller always passes in Strings that contains only ASCII. And thus from the point of view of the callee, it might as well specific limit the start of the identifier to be [a-zA-Z_]

Reply:
Hm, but `Row` is (was?) not limited to have ASCII-only field names:

```
presto> select cast(row(1) as row("łąka" integer));
   _col0
------------
 {"łąka"=1}
(1 row)
```

Reply:
An (un-delimited) Identifier has to match "[a-zA-Z_]([a-zA-Z0-9_:@])*", while a delimited identifier can have any characters within the quote. In the example you provided, the field name is quoted and thus is indeed allow.

In `parseRowTypeSignature()`, when state = 'START_OF_FIELD' and a quote (") is encountered, it will state-transitioned to `DELIMITED_NAME`, in which there is no restriction to the characters within. Hence, `isValidStartOfIdentifier(c)` is verifying the validity of the first character if an un-delimited identifier.

Reply:
To make the argument more obvious, you can also have `row("0123" integer)`.

Reply:
I think we should extend our test case to include non-ascii

Reply:
@caithagoras @haozhun thank you both for the explanation. I didn't realize this code is used for unquoted identifiers only.